BDArmory/UI/BDTeamSelector.cs

```diff
@@ -49,8 +49,14 @@ void SetVisible(bool visible)
         private void TeamSelectorWindow(int id)
         {
             height = margin;
+            GUI.Label(new Rect(margin, height, width - 2 * margin, buttonHeight), StringUtils.Localize("#LOC_BDArmory_SelectTeam"), BDArmorySetup.BDGuiSkin.label);
```

Comment:
Maybe add a drag region and hide the window when the VS gets hidden? Also, the window ID should be unique. This diff should cover these, fix a spelling mistake and also make the allies section auto-adjust to the correct height:
```diff
diff --git a/BDArmory/UI/BDTeamSelector.cs b/BDArmory/UI/BDTeamSelector.cs
index 88f4b8ae3..9e283e282 100644
--- a/BDArmory/UI/BDTeamSelector.cs
+++ b/BDArmory/UI/BDTeamSelector.cs
@@ -19,3 +19,3 @@ public class BDTeamSelector : MonoBehaviour
         const float buttonGap = 2;
-        const float newTeanButtonWidth = 40;
+        const float newTeamButtonWidth = 40;
         const float scrollWidth = 20;
@@ -29,2 +29,4 @@ public class BDTeamSelector : MonoBehaviour
         private Vector2 scrollPosition = Vector2.zero;
+        Rect alliesRect;
+        GUIStyle alliesStyle;
 
@@ -36,3 +38,2 @@ public void Open(MissileFire weaponManager, Vector2 position)
         {
-            SetVisible(true);
             targetWeaponManager = weaponManager;
@@ -40,5 +41,6 @@ public void Open(MissileFire weaponManager, Vector2 position)
             windowLocation = position;
+            SetVisible(true);
         }
 
-        void SetVisible(bool visible)
+        public void SetVisible(bool visible)
         {
@@ -46,2 +48,12 @@ void SetVisible(bool visible)
             GUIUtils.SetGUIRectVisible(guiCheckIndex, visible);
+            if (visible)
+            {
+                window = new Rect(
+                    Mathf.Min(windowLocation.x, Screen.width - (scrollable ? width + scrollWidth : width)),
+                    Mathf.Min(windowLocation.y, Screen.height - height),
+                    width,
+                    scrollable ? Screen.height / 2 + buttonHeight + buttonGap + 2 * margin : height
+                );
+                alliesStyle ??= new GUIStyle(BDArmorySetup.BDGuiSkin.label) { wordWrap = true, alignment = TextAnchor.UpperLeft, fixedWidth = width };
+            }
         }
@@ -52,2 +64,3 @@ private void TeamSelectorWindow(int id)
             GUI.Label(new Rect(margin, height, width - 2 * margin, buttonHeight), StringUtils.Localize("#LOC_BDArmory_SelectTeam"), BDArmorySetup.BDGuiSkin.label);
+            GUI.DragWindow(new Rect(margin, margin, width - 2 * margin - buttonHeight, buttonHeight));
             if (GUI.Button(new Rect(width - 18, 2, 18, 18), "X"))
@@ -58,6 +71,6 @@ private void TeamSelectorWindow(int id)
             // Team input field
-            newTeamName = GUI.TextField(new Rect(margin, height, width - buttonGap - 2 * margin - newTeanButtonWidth, buttonHeight), newTeamName, 30);
+            newTeamName = GUI.TextField(new Rect(margin, height, width - buttonGap - 2 * margin - newTeamButtonWidth, buttonHeight), newTeamName, 30);
 
             // New team button
-            Rect newTeamButtonRect = new Rect(width - margin - newTeanButtonWidth, height, newTeanButtonWidth, buttonHeight);
+            Rect newTeamButtonRect = new Rect(width - margin - newTeamButtonWidth, height, newTeamButtonWidth, buttonHeight);
             if (GUI.Button(newTeamButtonRect, StringUtils.Localize("#LOC_BDArmory_Generic_New"), BDArmorySetup.BDGuiSkin.button))//"New"
@@ -127,4 +140,6 @@ private void TeamSelectorWindow(int id)
             string allies = string.Join("; ", targetWeaponManager.Team.Allies);
-            GUI.Label(new Rect(margin, height, width - 2 * margin, buttonHeight), allies);
-            height += buttonHeight + buttonGap * Mathf.Clamp(Mathf.CeilToInt(allies.Length / 50), 1, 4);
+            alliesRect = new Rect(margin, height, width - 2 * margin, buttonHeight)
+            { height = alliesStyle.CalcHeight(new GUIContent(allies), width) };
+            GUI.Label(alliesRect, allies, alliesStyle);
+            height += alliesRect.height;
             if (scrollable)
@@ -147,2 +162,3 @@ private void TeamSelectorWindow(int id)
             height += margin;
+            window.height = scrollable ? Screen.height / 2 + buttonHeight + buttonGap + 2 * margin : height;
             GUIUtils.RepositionWindow(ref window);
@@ -164,9 +180,4 @@ protected virtual void OnGUI()
                 {
-                    var clientRect = new Rect(
-                        Mathf.Min(windowLocation.x, Screen.width - (scrollable ? width + scrollWidth : width)),
-                        Mathf.Min(windowLocation.y, Screen.height - height),
-                        width,
-                        scrollable ? Screen.height / 2 + buttonHeight + buttonGap + 2 * margin : height);
-                    if (BDArmorySettings.UI_SCALE != 1) GUIUtility.ScaleAroundPivot(BDArmorySettings.UI_SCALE * Vector2.one, clientRect.position);
-                    window = GUI.Window(10591029, clientRect, TeamSelectorWindow, "", BDArmorySetup.BDGuiSkin.window);
+                    if (BDArmorySettings.UI_SCALE != 1) GUIUtility.ScaleAroundPivot(BDArmorySettings.UI_SCALE * Vector2.one, window.position);
+                    window = GUI.Window(GUIUtility.GetControlID(FocusType.Passive), window, TeamSelectorWindow, "", BDArmorySetup.BDGuiSkin.window);
                     GUIUtils.UpdateGUIRect(window, guiCheckIndex);
diff --git a/BDArmory/UI/LoadedVesselSwitcher.cs b/BDArmory/UI/LoadedVesselSwitcher.cs
index 884ba9154..036cabb59 100644
--- a/BDArmory/UI/LoadedVesselSwitcher.cs
+++ b/BDArmory/UI/LoadedVesselSwitcher.cs
@@ -323,2 +323,3 @@ public void SetVisible(bool visible)
             GUIUtils.SetGUIRectVisible(_guiCheckIndex, visible);
+            if (!visible && BDTeamSelector.Instance) BDTeamSelector.Instance.SetVisible(false);
         }
@@ -896,3 +897,3 @@ void AddVesselSwitcherWindowEntry(MissileFire wm, string team, float height, flo
                 {
-                    BDTeamSelector.Instance.Open(wm, new Vector2(Input.mousePosition.x, Screen.height - Input.mousePosition.y));
+                    BDTeamSelector.Instance.Open(wm, new Vector2(Input.mousePosition.x + 2 * _buttonHeight + _margin, Screen.height - Input.mousePosition.y));
                 }
```